diff --git a/mbedtls-sys/vendor/library/aesni.c b/mbedtls-sys/vendor/library/aesni.c
index 062708b..2b1b29d 100644
--- a/mbedtls-sys/vendor/library/aesni.c
+++ b/mbedtls-sys/vendor/library/aesni.c
@@ -48,6 +48,7 @@
 
 #if defined(MBEDTLS_HAVE_X86_64)
 
+#if !defined(MBEDTLS_CUSTOM_HAS_AESNI)
 /*
  * AES-NI support detection routine
  */
@@ -68,6 +69,7 @@ int mbedtls_aesni_has_support( unsigned int what )
 
     return( ( c & what ) != 0 );
 }
+#endif
 
 /*
  * Binutils needs to be at least 2.19 to support AES-NI instructions.
diff --git a/mbedtls-sys/vendor/library/padlock.c b/mbedtls-sys/vendor/library/padlock.c
index b85ff9c..702bc76 100644
--- a/mbedtls-sys/vendor/library/padlock.c
+++ b/mbedtls-sys/vendor/library/padlock.c
@@ -43,6 +43,7 @@
 
 #if defined(MBEDTLS_HAVE_X86)
 
+#if !defined(MBEDTLS_CUSTOM_HAS_PADLOCK)
 /*
  * PadLock detection routine
  */
@@ -73,6 +74,7 @@ int mbedtls_padlock_has_support( int feature )
 
     return( flags & feature );
 }
+#endif
 
 /*
  * PadLock AES-ECB block en(de)cryption
